File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/schedule/ScheduleListAdapter.java
package ru.sigil.fantasyradio.schedule;

import android.app.Dialog;
import android.content.Context;
import android.content.res.Resources;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.BaseExpandableListAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.nostra13.universalimageloader.core.ImageLoader;

import org.joda.time.LocalDate;
import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import java.util.ArrayList;

import ru.sigil.fantasyradio.R;

class ScheduleListAdapter extends BaseExpandableListAdapter {

    private String[] item;
    private Context context;

    private DateTimeFormatter fmt = DateTimeFormat.forPattern("HH':'mm");
    private ArrayList<ArrayList<ScheduleEntity>> entities;

    public ScheduleListAdapter(Context context,
                               ArrayList<ArrayList<ScheduleEntity>> objects) {
        this.context = context;
        this.entities = objects;
        Resources res = context.getResources();
        if (res != null)
            item = res.getStringArray(R.array.week);
        else
            item = new String[]{"", "", "", "", "", "", ""};
    }

    @Override
    public int getGroupCount() {
        return entities.size();
    }

    @Override
    public int getChildrenCount(int groupPosition) {
        return entities.get(groupPosition).size();
    }

    @Override
    public Object getGroup(int groupPosition) {
        return entities.get(groupPosition);
    }

    @Override
    public Object getChild(int groupPosition, int childPosition) {
        return entities.get(groupPosition).get(childPosition);
    }

    @Override
    public long getGroupId(int groupPosition) {
        return groupPosition;
    }

    @Override
    public long getChildId(int groupPosition, int childPosition) {
        return childPosition;
    }

    @Override
    public boolean hasStableIds() {
        return true;
    }

    @Override
    public View getGroupView(int groupPosition, boolean isExpanded,
                             View convertView, ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) this.context
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.group_view, null);
        }
        TextView textGroup = null;
        if (convertView != null) {
            textGroup = convertView
                    .findViewById(R.id.textGroup);
        }
        LocalDate ld = LocalDate.now();
        ld = ld.plusDays(groupPosition);
        textGroup.setText(item[ld.getDayOfWeek() - 1] + ", "
                + ld.getDayOfMonth() + "." + ld.getMonthOfYear());
        return convertView;

    }

    public View getChildView(int groupPosition, int childPosition,
                             boolean isLastChild, View convertView, ViewGroup parent) {
        View row = convertView;
        if (row == null) {
            LayoutInflater inflater = (LayoutInflater) this.context
                    .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            row = inflater.inflate(R.layout.schedule_list_item, parent, false);
        }
        final ScheduleEntity message = entities.get(groupPosition).get(childPosition);
        TextView startTime = null;
        if (row != null) {
            startTime = row.findViewById(R.id.ScheduleItemStartTime);
        }
        startTime.setText(fmt.print(message.getStartDate()) + " - " + fmt.print(message.getEndDate()));
        TextView title = row.findViewById(R.id.ScheduleItemTitle);
        title.setText(message.getTitle());
        final String titleString = message.getTitle();
        final String messageString = message.getText();
        final String imageURLString = message.getImageURL();
        //----------------------------------------
        row.setOnClickListener(v -> {
            //Кликнули на элемент списка. Показываем окошко с подробностями
            final Dialog alertDialog = new Dialog(context);
            alertDialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
            alertDialog.setContentView(R.layout.schedule_dialog);
            LinearLayout ll = (alertDialog.findViewById(R.id.scheduleDialogLayout));
            ll.setOnClickListener(v1 -> alertDialog.cancel());
            LinearLayout ll2 = (alertDialog.findViewById(R.id.scheduleDialogInternalLayout));
            ll2.setOnClickListener(v12 -> alertDialog.cancel());
            TextView tv = (alertDialog.findViewById(R.id.schedule_dialog_text));
            tv.setText(messageString);
            ImageView im = (alertDialog.findViewById(R.id.schedule_dialog_image));
            TextView tv2 = (alertDialog.findViewById(R.id.schedule_dialog_title));
            tv2.setText(titleString);
            try {
                alertDialog.setCancelable(true);
                alertDialog.setCanceledOnTouchOutside(true);
                ImageLoader imageLoader = ImageLoader.getInstance();
                imageLoader.displayImage(imageURLString, im);
            } catch (Exception e) {
                e.printStackTrace();
            }
            alertDialog.show();
        });
        //------------------------------------------------------------
        return row;
    }

    @Override
    public boolean isChildSelectable(int groupPosition, int childPosition) {
        return true;
    }
}

File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/schedule/ScheduleListAdapter.kt
package ru.sigil.fantasyradio.schedule

import android.app.Dialog
import android.content.Context
import android.widget.BaseExpandableListAdapter
import org.joda.time.format.DateTimeFormat
import ru.sigil.fantasyradio.R
import android.widget.TextView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import org.joda.time.LocalDate
import com.nostra13.universalimageloader.core.ImageLoader
import android.widget.LinearLayout
import android.view.Window
import android.widget.ImageView

/**
 * Created by namelessone
 * on 03.12.18.
 */
class ScheduleListAdapter(private val context: Context?, private val entities: ArrayList<ArrayList<ScheduleEntity>>): BaseExpandableListAdapter() {
    private val item: Array<String>
    private val fmt = DateTimeFormat.forPattern("HH':'mm")

    init {
        val res = context?.resources
        item = if (res != null)
            res.getStringArray(R.array.week)
        else
            arrayOf("", "", "", "", "", "", "")
    }

    override fun getGroupCount(): Int {
        return entities.size
    }

    override fun getChildrenCount(groupPosition: Int): Int {
        return entities[groupPosition].size
    }

    override fun getGroup(groupPosition: Int): Any {
        return entities[groupPosition]
    }

    override fun getChild(groupPosition: Int, childPosition: Int): Any {
        return entities[groupPosition][childPosition]
    }

    override fun getGroupId(groupPosition: Int): Long {
        return groupPosition.toLong()
    }

    override fun getChildId(groupPosition: Int, childPosition: Int): Long {
        return childPosition.toLong()
    }

    override fun hasStableIds(): Boolean {
        return true
    }

    override fun getGroupView(groupPosition: Int, isExpanded: Boolean,
                     convertView: View?, parent: ViewGroup): View? {
        var view = convertView
        if (view == null) {
            val inflater = context?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(R.layout.group_view, null)
        }
        val textGroup = view?.findViewById<TextView>(R.id.textGroup)
        var ld = LocalDate.now()
        ld = ld.plusDays(groupPosition)
        textGroup!!.text = (item[ld.dayOfWeek - 1] + ", " + ld.dayOfMonth + "." + ld.monthOfYear)
        return view
    }

    override fun getChildView(groupPosition: Int, childPosition: Int,
                              isLastChild: Boolean, convertView: View, parent: ViewGroup): View {
        var row: View? = convertView
        if (row == null) {
            val inflater = context?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            row = inflater.inflate(R.layout.schedule_list_item, parent, false)
        }
        val (startDate, endDate, titleString, imageURLString, messageString) = entities[groupPosition][childPosition]
        var startTime: TextView? = null
        if (row != null) {
            startTime = row.findViewById(R.id.ScheduleItemStartTime)
        }
        startTime!!.text = fmt.print(startDate) + " - " + fmt.print(endDate)
        val title = row!!.findViewById<TextView>(R.id.ScheduleItemTitle)
        title.text = titleString
        row.setOnClickListener { _ ->
            //Кликнули на элемент списка. Показываем окошко с подробностями
            val alertDialog = Dialog(context)
            alertDialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
            alertDialog.setContentView(R.layout.schedule_dialog)
            val ll = alertDialog.findViewById<LinearLayout>(R.id.scheduleDialogLayout)
            ll.setOnClickListener({ _ -> alertDialog.cancel() })
            val ll2 = alertDialog.findViewById<LinearLayout>(R.id.scheduleDialogInternalLayout)
            ll2.setOnClickListener({ _ -> alertDialog.cancel() })
            val tv = alertDialog.findViewById<TextView>(R.id.schedule_dialog_text)
            tv.text = messageString
            val im = alertDialog.findViewById<ImageView>(R.id.schedule_dialog_image)
            val tv2 = alertDialog.findViewById<TextView>(R.id.schedule_dialog_title)
            tv2.text = titleString
            try {
                alertDialog.setCancelable(true)
                alertDialog.setCanceledOnTouchOutside(true)
                val imageLoader = ImageLoader.getInstance()
                imageLoader.displayImage(imageURLString, im)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            alertDialog.show()
        }
        return row
    }

    override fun isChildSelectable(groupPosition: Int, childPosition: Int): Boolean {
        return true
    }
}